Feat: Torchserve v2 api

jagadeeshi2i opened the pull request:
- Adds v2 API for Torchserve
- Adds gRPC support for Torchserve

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1253 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note

```

Commit:
feat: v2 protocol support for torchserve

Commit:
feat: v2 protocol support for torchserve

Commit:
feat: v2 protocol support for torchserve

Commit:
feat: v2 protocol support for torchserve

Commit:
Feat: Add unit test for torchserve predictor

 - Add e2e test

Commit:
Feat: Add unit test for torchserve predictor

 - Add e2e test

Commit:
Feat: Add unit test for torchserve predictor

 - Add e2e test

Commit:
Feat: Add unit test for torchserve predictor

 - Add e2e test

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add readme doc for v2 support

 - fix lint error
 - add grpc sample yaml
 - add tensor input generation script
 - fix model archiver to support v2 protocol

Commit:
Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Commit:
Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Commit:
Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Commit:
Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Commit:
Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Commit:
Feat: Add gRPC client

 - fix lint error
 - update readme for gRPC

Commit:
fix: lint error

Commit:
fix: lint error

Commit:
fix: lint error

Commit:
fix: lint error

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
fix: Remove grpc proto files

 - fix readme doc

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
feat: Add custom handlers for v2 api

Commit:
fix: lint errors

Commit:
fix: lint errors

Commit:
fix: lint errors

Commit:
fix: lint errors

Commit:
fix: lint errors

Commit:
fix: lint errors

Commit:
fix: lint errors

Commit:
fix: e2e test

Commit:
fix: e2e test

Commit:
fix: e2e test

Commit:
fix: e2e test

Commit:
fix: manifest crd

Commit:
fix: manifest crd

Commit:
fix: manifest crd

Commit:
fix: manifest crd

Commit:
fix: e2e test for batcher and torchserve

Commit:
fix: e2e test for batcher and torchserve

Commit:
fix: e2e test for batcher and torchserve

Commit:
fix: e2e test for batcher and torchserve

yuzisun commented:
/retest

yuzisun committed:
Update doc pointers to KServe website (#1831)

* Update docs

* Update pointer for creating first isvc

zijianjoy committed:
(GKE): Update documentation about calling inference service endpoing behind IAP. Fix https://github.com/kubeflow/gcp-blueprints/issues/324 (#1832)

* (GKE): Update documentation about calling inference service endpoing behind IAP

* KFserve migration

* updated to v1beta1

* lint issue

yuzisun committed:
Update torchserve image in test overlay

yuzisun committed:
Update torchserve image in test overlay

yuzisun committed:
Update torchserve image in test overlay

yuzisun committed:
Update torchserve image in test overlay

yuzisun commented:
@jagadeeshi2i  we need to deprecate the original pytorch test, can you also add a release note for the deprecation plan?

Commit:
fix: torchserve test

 - fix readme

Commit:
fix: torchserve test

 - fix readme

Commit:
fix: torchserve test

 - fix readme

Commit:
fix: torchserve test

 - fix readme

Commit:
fix: skip pytorchserver test

 - fix transformer test

Commit:
fix: skip pytorchserver test

 - fix transformer test

Commit:
fix: skip pytorchserver test

 - fix transformer test

Commit:
fix: skip pytorchserver test

 - fix transformer test

Commit:
fix: pylint error

Commit:
fix: pylint error

Commit:
fix: pylint error

Commit:
fix: pylint error

Commit:
fix: transformer test

Commit:
fix: transformer test

Commit:
fix: transformer test

Commit:
fix: transformer test

Commit:
fix: transformer test

Commit:
fix: transformer test

Commit:
fix: add deprecation notice for pytorch server

Commit:
fix: add deprecation notice for pytorch server

Commit:
fix: add deprecation notice for pytorch server

Commit:
fix: add deprecation notice for pytorch server

Commit:
fix: update readme doc with v2 support for torchserve

Commit:
fix: update readme doc with v2 support for torchserve

Commit:
fix: update readme doc with v2 support for torchserve

Commit:
fix: update readme doc with v2 support for torchserve

Commit:
Update Triton to 21.09. Remove caffe2 from supported frameworks (#1840)

yuzisun committed:
Azure Blob Upgrade Bug Fix #1750 (#1845)

* switched from list_blobs to walk_blobs


correction with prefix


fixed lint error

* corrected test cases

accounting for walk_blob calls and multiple stack calls

* Add lstrip

* Fix test_deep_blob test

Co-authored-by: Luis <maganaluis92@gmail.com>

Iamlovingit committed:
add hpa for raw deployment (#1830)

* add autoscaler hpa

* update validation

* add test code

* add behavior default value

* add validation for target utilization percentage

yuzisun committed:
Update KServe main README to redirect to website (#1847)

* Update KServe main readme

yuzisun committed:
Update README.md

pvaneck committed:
Allow unknown fields in InferenceService status (#1855)

This will allow various status fields populated by ModelMesh Serving
to appear under the custom resource status.

Iamlovingit committed:
fix-1852: when minReplicas is nil, make the controller crash. (#1854)

* fix-1852: when minReplicas is nil, make the controller crush.

Co-authored-by: Dan Sun <dsun20@bloomberg.net>

Suresh-Nakkeran committed:
adding tgz/zip support for all storages (#1836)

* adding tgz/zip support for all storages

* Fix lint

Co-authored-by: Dan Sun <dsun20@bloomberg.net>

ScrapCodes committed:
Do not reconcile if modelmesh annotation set as true. (#1851)

jeremy-london committed:
Updates README.md (#1856)

Co-authored-by: London, Jeremy <jeremy.london@lmco.com>

Suresh-Nakkeran committed:
Migration job (#1834)

* Adding migration job for isvc migration from kfserving

* remove revision clean up step and ignore webapp from validation

* verify inferenceservice status before and after migration

yuzisun commented on docs/samples/README.md:
```diff
@@ -26,14 +26,14 @@ After models are deployed onto model servers with KFServing, you get all the fol
 | [Triton Inference Server](https://github.com/triton-inference-server/server) | [TensorFlow,TorchScript,ONNX,TensorRT](https://docs.nvidia.com/deeplearning/triton-inference-server/user-guide/docs/model_repository.html)| v2 | :heavy_check_mark: | :heavy_check_mark: | [Compatibility Matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html)| [Triton Examples](./v1beta1/trit ...
 | [TFServing](https://www.tensorflow.org/tfx/guide/serving) | [TensorFlow SavedModel](https://www.tensorflow.org/guide/saved_model) | v1 | :heavy_check_mark: | :heavy_check_mark: | [TFServing Versions](https://github.com/tensorflow/serving/releases) | [TensorFlow Examples](./v1alpha2/tensorflow)  |
-| [TorchServe](https://pytorch.org/serve/server.html) | [Eager Model/TorchScript](https://pytorch.org/docs/master/generated/torch.save.html) | v1 | :heavy_check_mark: | :heavy_check_mark: | 0.4.1 | [TorchServe Examples](./v1beta1/torchserve)  |
+| [TorchServe](https://pytorch.org/serve/server.html) | [Eager Model/TorchScript](https://pytorch.org/docs/master/generated/torch.save.html) | v2 | :heavy_check_mark: | :heavy_check_mark: | 0.4.1 | [TorchServe Examples](./v1beta1/torchserve)  |
```
```suggestion
| [TorchServe](https://pytorch.org/serve/server.html) | [Eager Model/TorchScript](https://pytorch.org/docs/master/generated/torch.save.html) | v1/v2 | :heavy_check_mark: | :heavy_check_mark: | 0.4.1 | [TorchServe Examples](./v1beta1/torchserve)  |
```

zijianjoy committed:
Update docs/samples/gcp-iap/README.md (#1860)

animeshsingh committed:
Update README.md (#1861)

andyi2it committed:
Deprecation warnings for v1beta1 webhook and crd (#1837)

* Initial commit of deprecarion changes

* Updated apiextensions to v1 from v1beta1

* Added a script to fix issue with protocol default value.
Updated CRD.

* Updated manifests

* Updated params, webhook manifest and conversion webhook.

* Reverting test crds

* Updated fieldpath typo in kustomization.yaml

* Updated manifests.

* Updated make file to exclude v1alpha2 package for make test

Added a debug log, will be removed once debug is complete

Made changes to test script to update kustomize

Updated e2e script to properly install kustomize

Updated kustomize version

Fixed an issue with quick install

Fixed an issue with paddle and canary rollout.

* Remove the duplicated webhook configuration

* Fixed issue with webhook manifest.

Co-authored-by: Dan Sun <dsun20@bloomberg.net>

Suresh-Nakkeran committed:
deprecate v1alpha2 api group (#1850)

* deprecate v1alpha2 api group

* update transformer e2e test to use v1beta1

* Update test_transformer.py

* fix model name in test

* Fix the assert

Co-authored-by: Dan Sun <dsun20@bloomberg.net>

yuzisun committed:
Publish KServe 0.7.0 release (#1848)

* Generate 0.7.1 release

* Update SDK to 0.7.0rc1

* Delete kubeflow crd manifests

* Generate  0.7.0 release manifests

* Update sdk version to 0.7.0

andyi2it committed:
Fixed an issue with quick install script. (#1862)

Fixed an issue with make file which breaks the CRDs.
Regerenated the release yamls.

yuzisun committed:
Update model server versions

yuzisun committed:
Quick install fix and fix kubeflow manifest image versions (#1865)

* Update version to 0.7.0 in configuration

* Fix sklearn perf test

* Fix version

yuzisun committed:
Update KFSERVING_DEBUG_GUIDE.md

Commit:
fix: readme doc

Commit:
fix: readme doc

Commit:
fix: readme doc

Commit:
Merge branch 'torchserve-v2-api' of https://github.com/jagadeeshi2i/kserve into torchserve-v2-api

Commit:
Revert "Merge branch 'torchserve-v2-api' of https://github.com/jagadeeshi2i/kserve into torchserve-v2-api"

This reverts commit 74703f6fa57a6b21a8690e50cee1865f541d179c, reversing
changes made to a58fd26fdb8ef0132c5d35a36272571429ae4601.

Commit:
Revert "Merge branch 'torchserve-v2-api' of https://github.com/jagadeeshi2i/kserve into torchserve-v2-api"

This reverts commit 74703f6fa57a6b21a8690e50cee1865f541d179c, reversing
changes made to a58fd26fdb8ef0132c5d35a36272571429ae4601.

Commit:
Revert "Merge branch 'torchserve-v2-api' of https://github.com/jagadeeshi2i/kserve into torchserve-v2-api"

This reverts commit 74703f6fa57a6b21a8690e50cee1865f541d179c, reversing
changes made to a58fd26fdb8ef0132c5d35a36272571429ae4601.

Commit:
Merge branch 'torchserve-v2-api' of https://github.com/jagadeeshi2i/kserve into torchserve-v2-api

Commit:
Merge branch 'torchserve-v2-api' of https://github.com/jagadeeshi2i/kserve into torchserve-v2-api

jagadeeshi2i commented:
PR Moved to - https://github.com/kserve/kserve/pull/1870
